guides/keras_cv/generate_with_stable_diffusion.py

```diff
@@ -0,0 +1,294 @@
+"""
+Title: Generate images using KerasCV's StableDiffusion's at unprecedented speeds
+Author: [divamgupta](https://github.com/divamgupta), [fchollet](https://github.com/fchollet), [lukewood](https://lukewood.xyz), [ianstenbit](https://github.com/ianstenbit), ... and others if anyone wants to contribute
+Date created: 2022/09/24
+Last modified: 2022/09/24
+Description:
+"""
+
+"""
+## Overview
+
+In this guide, we will show how to generate novel images based on a text prompt using
+the KerasCV implementation of [stability.ai's](https://stability.ai/) image to text
+model,
+[StableDiffusion](https://github.com/CompVis/stable-diffusion).
+
+StableDiffusion is a powerful, open-source text to image generation model.  While there
+exist numerous open source implementations that allow you to easily create images from
+textual prompts, KerasCV's offers a few distinct advantages.
+These include [XLA compilation](https://www.tensorflow.org/xla) and
+[mixed precision computation](https://www.tensorflow.org/guide/mixed_precision).
+
+In this guide, we will explore KerasCV's StableDiffusion implementation, show how to use
+these powerful performance boosts, and explore the performance benefits
+that they offer.
+
+To get started, lets install a few dependencies and sort out some imports:
+"""
+
+"""shell
+pip install ez-timer luketils
+"""
+
+import keras_cv
+from luketils import visualization
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+from tensorflow.keras import mixed_precision
+from ez_timer import ez_timer
+
+"""
+## Introduction
+
+Unlike most tutorials, where we first explain a topic then show how to implement it,
+with text to image generation it is easiest to show instead of tell.
+
+Check out the power of `keras_cv.models.StableDiffusion()`.
+First, we construct a model:
+"""
+
+stable_diffusion = keras_cv.models.StableDiffusion()
+
+"""
+Next, we give it a prompt:
+"""
+
+
+def visualize_prompt(prompt, sd_model=None):
+    sd_model = sd_model or stable_diffusion
+    visualization.plot_gallery(
+        sd_model.text_to_image(prompt, batch_size=3),
+        rows=1,
+        cols=3,
+        scale=4,
+        value_range=(0, 255),
+        show=True,
+    )
+
+
+visualize_prompt("a cartoon caterpillar wearing glasses")
+
+"""
+
+Pretty incredible!
+
+But that's not all this model can do.
+Have you ever seen a monkey surf?
+"""
+
+visualize_prompt(
+    "The worlds cutest monkey surfing on a " "beautiful wave in the sunshine!"
+)
+
+"""
+Or the Garden of Eden filled with cats?
+"""
+
+visualize_prompt(
+    "Absolutely stunning artwork of the garden of eden, " "but filled with cats!"
+)
+
+"""
+Or a cute magical flying dog with a lengthy list of descriptors:
+"""
+
+
+visualize_prompt(
```

Comment:
IMO there's no need to have tons of prompts right at the start, because we have plenty of opportunities to show off prompts while we intro MP, XLA, etc.